### ISSUE TITLE
length field incorrect for @PolyDet arrays

### ISSUE THREAD

**Issue by jwaataja** [t-rasmud/checker-framework#60]
The following code currently type checks:
```java
import org.checkerframework.checker.determinism.qual.*;

public class ArrayLen {
  public static void f(@PolyDet int @PolyDet [] a) {
    @Det int len = a.length;
  }
}
```

This is invalid because `a` could have the type `@NonDet int @NonDet []` which would have non-deterministic length. The correct type would be `@PolyDet("down")` as with lists.

This only applies to `@PolyDet`. The correct error appears when using `@NonDet` directly:
```java
import org.checkerframework.checker.determinism.qual.*;

public class ArrayLen {
  public static void f(@NonDet int @NonDet [] a) {
    @Det int len = a.length;
  }
}
```
Error:
```
ArrayLen.java:5: error: [assignment.type.incompatible] incompatible types in assignment.
    @Det int len = a.length;
                    ^
  found   : @NonDet int
  required: @Det int
```

**Comment by t-rasmud** [comment]
Fixed in bffa1db1fe00cc88d3878cdb0b450a3a9eb3db8b

**Comment by jwaataja** [comment]
It no longer returns `@Det`, but now `@PolyDet("up")`. The correct type is `@PolyDet("down")`. The error from the above code is now:
```
    @Det int len = a.length;
                    ^
  found   : @PolyDet("up") int
  required: @Det int
```
